File: plugin/integration/dataviz-plugin/src/scala/org/jetbrains/zeppelin/dataviz/DatavizOutputHandler.scala
package org.jetbrains.zeppelin.dataviz

import jetbrains.zeppelin.service.TableOutputHandler
import jetbrains.zeppelin.utils.ZeppelinLogger
import org.intellij.datavis.data.DataProvider

class DatavizOutputHandler extends TableOutputHandler {
  override def invoke(msg: String): Unit = {
    val id = "zeppelin-result"
    val name = "result"
    val dataProvider: DataProvider = DataProvider.INSTANCE
    try {
      if (dataProvider.isExist(id)) {
        dataProvider.removeData(id)
      }
      dataProvider.addData(id, name, msg, '\t')
    }
    catch {
      case e: Throwable => {
        ZeppelinLogger.printError(s"Cannot perform show table in dataviz plugin. ${e.getMessage}")
      }
    }
  }
}

File: plugin/integration/dataviz-plugin/src/scala/org/jetbrains/zeppelin/dataviz/DatavizOutputHandler.java
package org.jetbrains.zeppelin.dataviz;

import com.intellij.openapi.project.Project;
import jetbrains.zeppelin.service.TableOutputHandler;
import jetbrains.zeppelin.utils.ZeppelinLogger;
import org.intellij.datavis.data.DataProvider;
import org.intellij.datavis.data.DataProviderWrapper;

public class DatavizOutputHandler implements TableOutputHandler {
    @Override
    public void invoke(Project project, String msg) {
        String id = "Zeppelin";
        String name = "Result";
        DataProvider dataProvider = DataProviderWrapper.getProvider(project);
        try {
            if (dataProvider.isExist(id)) {
                dataProvider.removeData(id);
            }
            dataProvider.addData(id, name, msg, '\t');
        } catch (Exception e) {
            ZeppelinLogger.printError("Cannot perform show table in dataviz plugin");
        }
    }
}